cmd/kubectl-kubestellar/cmd/ensure/wds.go

```diff
@@ -0,0 +1,144 @@
+func newCmdEnsureWds(cliOpts *genericclioptions.ConfigFlags) *cobra.Command {
+	// Make wds command
+	cmdWds := &cobra.Command{
+		Use:     "wds <WDS_NAME> --with-kube=<TRUE/FALSE>",
+		Aliases: []string{"wmw"},
+		Short:   "Ensure existence and configuration of a workload description space (WDS, formerly WMW)",
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// At this point set silence usage to true, so that any errors
+			// following do not result in the help being printed. We only
+			// want the help to be displayed when the error is due to an
+			// invalid command.
+			cmd.SilenceUsage = true
+			err := ensureWds(cmd, args, cliOpts)
+			return err
+		},
+	}
+
+	// Add flag for
+	cmdWds.Flags().BoolVar(&withKube, "with-kube", true, "Include root:compute API bindings")
+	cmdWds.MarkFlagRequired("with-kube")
+	return cmdWds
+}
+
+// Perform validation of workload description space (WDS). The user will provide
+// the WDS name as a command-line argument, along with a boolean for --with-kube.
+// This function will:
+//   - At the root KCP level, check for a WDS workspace having the user provided
+//     name, create if needed
+//   - At the root KCP level, check for an APIBinding "bind-espw" with export path
+//     "root:espw" and export name "edge.kubestellar.io"
+//   - If --with-kube is true, ensure a list of APIBindings exist with export path
+//     "root:compute" (create any that are missing). If --with-kube is false, make
+//     sure none of these exist (delete as needed).
+func ensureWds(cmdWds *cobra.Command, args []string, cliOpts *genericclioptions.ConfigFlags) error {
+	wdsName := args[0] // name of WDS
+	ctx := context.Background()
+	logger := klog.FromContext(ctx)
+
+	// Print all flags and their values if verbosity level is at least 1
+	cmdWds.Flags().VisitAll(func(flg *pflag.Flag) {
+		logger.V(1).Info(fmt.Sprintf("Command line flag %s=%s", flg.Name, flg.Value))
+	})
+
+	// Make sure user provided WDS name is valid
+	err := plugin.CheckWdsName(wdsName)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("Invalid WDS name %s", wdsName))
+		return err
+	}
+
+	// Set context to root, later on we will append the WDS name to the root server
+	configContext := "root"
+	cliOpts.Context = &configContext
+
+	// Get client config from flags
+	config, err := cliOpts.ToRESTConfig()
+	if err != nil {
+		logger.Error(err, "Failed to get config from flags")
+		return err
+	}
+
+	// Create client-go instance from config
+	client, err := kcpclientset.NewForConfig(config)
+	if err != nil {
+		logger.Error(err, "Failed create client-go instance")
+		return err
+	}
+
+	// Check for WDS workspace, create if it does not exist
+	// This function prints its own log messages, so no need to add any here.
+	err = plugin.VerifyOrCreateWDS(client, ctx, wdsName)
+	if err != nil {
+		return err
+	}
+
+	// Update host to work on objects within WDS workspace
+	config.Host += ":" + wdsName
```

Comment:
`config` here is a _pointer_ to a data structure that is already in use from the earlier `NewForConfig()`, right? I think it would be safer to copy the Config struct and modify the copy.